extensions/data-transfer/portability-data-transfer-flickr/src/main/java/org/dataportabilityproject/datatransfer/flickr/photos/FlickrPhotosExporter.java

```diff
@@ -0,0 +1,225 @@
+  private static final String PHOTOSET_EXTRAS = "";
+
+  private final PhotosetsInterface photosetsInterface;
+  private final PhotosInterface photosInterface;
+  private final Flickr flickr;
+
+  private final Logger logger = LoggerFactory.getLogger(FlickrPhotosExporter.class);
+
+  public FlickrPhotosExporter(AppCredentials appCredentials) {
+    this.flickr = new Flickr(appCredentials.getKey(), appCredentials.getSecret(), new REST());
+    this.photosetsInterface = flickr.getPhotosetsInterface();
+    this.photosInterface = flickr.getPhotosInterface();
+  }
+
+  @VisibleForTesting
+  FlickrPhotosExporter(Flickr flickr) {
+    this.flickr = flickr;
+    this.photosInterface = flickr.getPhotosInterface();
+    this.photosetsInterface = flickr.getPhotosetsInterface();
+  }
+
+  @VisibleForTesting
+  static PhotoModel toCommonPhoto(Photo p, String albumId) {
+    Preconditions.checkArgument(
+        !Strings.isNullOrEmpty(p.getOriginalSize().getSource()),
+        "Photo [" + p.getId() + "] has a null authUrl");
+    return new PhotoModel(
+        p.getTitle(),
+        p.getOriginalSize().getSource(),
+        p.getDescription(),
+        toMimeType(p.getOriginalFormat()),
+        albumId);
+  }
+
+  @VisibleForTesting
+  static String toMimeType(String flickrFormat) {
+    switch (flickrFormat) {
+      case "jpg":
+      case "jpeg":
+        return "image/jpeg";
+      default:
+        throw new IllegalArgumentException("Don't know how to map: " + flickrFormat);
+    }
+  }
+
+  @Override
+  public ExportResult<PhotosContainerResource> export(AuthData authData) {
+    return export(authData, new ExportInformation(null, null));
+  }
+
+  @Override
+  public ExportResult<PhotosContainerResource> export(
+      AuthData authData, ExportInformation exportInformation) {
+
+    PaginationData paginationData = exportInformation.getPaginationData();
+
+    IdOnlyContainerResource resource =
+        (IdOnlyContainerResource) exportInformation.getContainerResource();
+    if (resource != null) {
+      return getPhotos(resource, paginationData);
+    } else {
+      return getAlbums(paginationData, getAuth(authData));
+    }
+  }
+
+  private ExportResult<PhotosContainerResource> getPhotos(
+      IdOnlyContainerResource resource, PaginationData paginationData) {
+    String photoSetId = resource.getId();
+    int page = paginationData == null ? 1 : ((IntPaginationToken) paginationData).getStart();
+    PhotoList<Photo> photoSetList;
+
+    try {
+      if (photoSetId == null) {
+        RequestContext.getRequestContext().setExtras(EXTRAS);
+        photoSetList = photosInterface.getNotInSet(PHOTO_PER_PAGE, page);
+        RequestContext.getRequestContext().setExtras(ImmutableList.of());
+      } else {
+        photoSetList =
+            photosetsInterface.getPhotos(
+                photoSetId, ImmutableSet.copyOf(EXTRAS), 0, PHOTO_PER_PAGE, page);
+      }
+    } catch (FlickrException e) {
+      throw new IllegalArgumentException(e);
```

Comment:
Maybe add "Error getting flickr photos" to the exception message

Reply:
Switched to returning an error export result instead of throwing an exception